### ISSUE TITLE
SA1119 incorrectly reporting on pattern matching (is expression)

### ISSUE THREAD

**Issue by nbarbettini** [DotNetAnalyzers/StyleCopAnalyzers#2372]
Given this code:

```csharp
object obj = null;

if (!(obj is string foo))
{
    // do something with foo
}
```

SA1119 reports the parenthesis as unnecessary (they are necessary). The code fix removes the parenthesis, resulting in syntax that won't compile:

```csharp
if (! obj is string foo)
{
    // do something with foo
}
```

It would appear SA1119 is reaching for the proposed [is not](https://github.com/dotnet/csharplang/issues/246) or [aint](https://github.com/dotnet/csharplang/issues/27#issuecomment-279592692) operator. 😉

**Comment by nbarbettini** [comment]
This should be part of https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/2268 since it's caused by a new C# 7 feature (is-expression pattern matching).

**Comment by sharwell** [comment]
This particular pattern should already be covered by the light-up functionality in #2370, allowing someone to branch off that work and correct this issue.

**Comment by AdjutantML** [comment]
Thank you for fixing this

**Comment by Tvde1** [comment]
This is happening to me with StyleCop.Analyzers v1.0.2 on .net 4.7.2 latest C# version.
